pkg/kclient/volumes_test.go

```diff
@@ -0,0 +1,328 @@
+					}
+				}
+			}
+		})
+	}
+}
+
+func TestAddPVCToPodTemplateSpec(t *testing.T) {
+
+	container := &corev1.Container{
+		Name:            "container1",
+		Image:           "image1",
+		ImagePullPolicy: corev1.PullAlways,
+
+		Command: []string{"tail"},
+		Args:    []string{"-f", "/dev/null"},
+		Env:     []corev1.EnvVar{},
+	}
+
+	tests := []struct {
+		podName        string
+		namespace      string
+		serviceAccount string
+		pvc            string
+		volumeName     string
+		labels         map[string]string
+	}{
+		{
+			podName:        "podSpecTest",
+			namespace:      "default",
+			serviceAccount: "default",
+			pvc:            "mypvc",
+			volumeName:     "myvolume",
+			labels: map[string]string{
+				"app":       "app",
+				"component": "frontend",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.podName, func(t *testing.T) {
+
+			objectMeta := CreateObjectMeta(tt.podName, tt.namespace, tt.labels, nil)
+
+			podTemplateSpec := GeneratePodTemplateSpec(objectMeta, tt.serviceAccount, []corev1.Container{*container})
+
+			AddPVCToPodTemplateSpec(podTemplateSpec, tt.pvc, tt.volumeName)
+
+			pvcMatched := false
+			for _, volume := range podTemplateSpec.Spec.Volumes {
+				if volume.Name == tt.volumeName && volume.VolumeSource.PersistentVolumeClaim != nil && volume.VolumeSource.PersistentVolumeClaim.ClaimName == tt.pvc {
+					pvcMatched = true
+				}
+			}
+
+			if !pvcMatched {
+				t.Errorf("Volume does not exist with Volume Name %s and PVC claim name %s", tt.volumeName, tt.pvc)
+			}
+
+		})
+	}
+}
+
+func TestAddVolumeMountToPodTemplateSpec(t *testing.T) {
+
+	container := &corev1.Container{
+		Name:            "container1",
+		Image:           "image1",
+		ImagePullPolicy: corev1.PullAlways,
+
+		Command: []string{"tail"},
+		Args:    []string{"-f", "/dev/null"},
+		Env:     []corev1.EnvVar{},
+	}
+
+	tests := []struct {
+		podName                string
+		namespace              string
+		serviceAccount         string
+		pvc                    string
+		volumeName             string
+		containerMountPathsMap map[string][]string
+		labels                 map[string]string
+	}{
+		{
+			podName:        "podSpecTest",
+			namespace:      "default",
+			serviceAccount: "default",
+			pvc:            "mypvc",
+			volumeName:     "myvolume",
+			containerMountPathsMap: map[string][]string{
+				container.Name: {"/tmp/path1", "/tmp/path2"},
+			},
+			labels: map[string]string{
+				"app":       "app",
+				"component": "frontend",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.podName, func(t *testing.T) {
+
+			objectMeta := CreateObjectMeta(tt.podName, tt.namespace, tt.labels, nil)
+
+			podTemplateSpec := GeneratePodTemplateSpec(objectMeta, tt.serviceAccount, []corev1.Container{*container})
+
+			_ = AddVolumeMountToPodTemplateSpec(podTemplateSpec, tt.volumeName, tt.pvc, tt.containerMountPathsMap)
```

Comment:
do we need the `_ =` assignment here?

Reply:
The function returns an error, so it should probably be checked here.

Reply:
I have updated the test, I am now testing the err condition as well. I missed it previously. I have also updated some test code to reuse a odo util function for labels.

Pls review at your convenience, thx!